### PR TITLE
GHA workflow to promote returntocorp/semgrep:canary to :latest

### PR DESCRIPTION
**Pull request by aryx**
test plan:
trigger manually from GHA dashboard
hope nothing breaks

### CODE REVIEW

**Comment by aryx** [comment]
context: https://www.notion.so/semgrep/returntocorp-semgrep-canary-docker-canary-45e4927fb1f847429f6064a96bf9dffd

**Review by tpetr — CHANGES_REQUESTED** [review]
The Docker pull-tag-push approach won’t work for multi-arch images. I can help you with this today

File: .github/workflows/promote-canary-to-latest.yml [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+      - uses: docker/login-action@v3
+        with:
+          username: ${{ secrets.DOCKER_USERNAME }}
+          password: ${{ secrets.DOCKER_PASSWORD }}
+      - run: |
+          if [[ "${{ inputs.confirmeed }}" == "true" ]]; then
+             docker pull returntocorp/semgrep:canary
+             docker tag  returntocorp/semgrep:canary returntocorp/semgrep:latest
+             docker push returntocorp/semgrep:latest
+          fi
```
Comment (jbergler) [inline .github/workflows/promote-canary-to-latest.yml]: I'd consider approaching the input like this (also note the spelling mistake in the variable name)

```suggestion      
          docker pull returntocorp/semgrep:canary
          docker tag  returntocorp/semgrep:canary returntocorp/semgrep:latest
          if [[ "${{ inputs.confirmed }}" == "true" ]]; then
             docker push returntocorp/semgrep:latest
          else
             echo "dry-run: not pushing tag upstream
          fi
```

**Comment by aryx** [comment]
closing in favor of https://github.com/returntocorp/semgrep/pull/8836#pullrequestreview-1645095607

**Pull request closed without merging**